### PR TITLE
Introduce the ability to pause queue from the UI

### PR DESCRIPTION
**Pull request by shayonj**
Considering sidekiq web UI allow to perform complex tasks with ease.
Having the ability to pause a queue during an emergency (or operational intense)
scenarios would come in quite handy.

This PR introduces a new button to under the "Actions" column on `sidekiq/queues` pages,
right next to the `Delete` button. Depending on the state of the queue, it will according
show the `Pause` or `Unpause` text, with appropriate form element that `POST`s to the same
endpoint. The endpoint logic is updated to handle the new cases. Added some relevant unit tests.

Adding mock `Sidekiq::Queue#pause!` and `Sidekiq::Queue#unpause!` functions to
the API, which I believe will be overridden by sidekiq pro.

Original request: https://github.com/mperham/sidekiq/issues/4343

**Examples**
image
image

**Open questions**
- [ ] `Sidekiq::Queue#pause!` is a feature on sidekiq pro. Showing the `pause` button in the UI on regular sidekiq could be misleading, since it would be a no-op? 
  - Maybe it should be hidden through some custom check/feature-flag-esque behavior? And/or ported to sidekiq pro?
- [ ] Added new copy to `en.yml`, translations would need to be accordingly handled?

### CODE REVIEW

**Comment by shayonj** [comment]
cc @mperham

File: web/locales/en.yml [hunk 1]
```diff
@@ -24,9 +24,13 @@ en: # <---- change this to your locale code
   ShowAll: Show All
   CurrentMessagesInQueue: Current jobs in <span class='title'>%{queue}</span>
   Delete: Delete
+  Pause: Pause
+  Unpause: Unpause
   AddToQueue: Add to queue
   AreYouSureDeleteJob: Are you sure you want to delete this job?
   AreYouSureDeleteQueue: Are you sure you want to delete the %{queue} queue?
+  AreYouSurePauseQueue: Are you sure you want to pause the %{queue} queue?
```
Comment (mperham) [inline web/locales/en.yml]: We don't need confirmation for these actions.

File: web/locales/en.yml [hunk 2]
```diff
@@ -24,9 +24,13 @@ en: # <---- change this to your locale code
   ShowAll: Show All
   CurrentMessagesInQueue: Current jobs in <span class='title'>%{queue}</span>
   Delete: Delete
+  Pause: Pause
```
Comment (mperham) [inline web/locales/en.yml]: Please put new strings at the bottom of the file so other languages can see an easy diff of which strings are missing.

File: lib/sidekiq/api.rb [hunk 3]
```diff
@@ -242,11 +242,19 @@ def size
       Sidekiq.redis { |con| con.llen(@rname) }
     end
 
-    # Sidekiq Pro overrides this
+    # Sidekiq Pro overrides pause?, pause! and unpause! functions
     def paused?
       false
     end
 
+    def pause!
```
Comment (mperham) [inline lib/sidekiq/api.rb]: What happens if the user isn't using Sidekiq Pro?  Should these methods really do nothing?  Should they raise an error or log a warning?  Should the button be disabled in the Web UI with an explanatory tooltip if `!defined?(::Sidekiq::Pro)`?

Comment (shayonj) [reply lib/sidekiq/api.rb]: That's good question, I left it under open question so we could discuss some possible options. I am currently of the opinion that we roll it out with minimal change to the regular sidekiq and web (backwards compatible) and as a feature on sidekiq pro. That means, we simply hide the button when pro is not present from the UI and accordingly add proper validation in the api router. Raising exception or tooltips would result in unnecessary friction that didn't exist before. Of course, can be persuaded :). 

I introduced this as a new commit. Its not a huge change. Used `defined?(::Sidekiq::Pro)` as a check to determine whether or not pro is present.

**Comment by mperham** [comment]
Thank you, I'm very happy with this.

**Comment by shayonj** [comment]
woot! thanks for the quick feedback and reviews 🙌

**Comment by jakemack** [comment]
Thanks for picking up my slack here, @shayonj. Awesome work!

**Pull request merged**